### PR TITLE
Get typing and completion commands working

### PR DESCRIPTION
**Pull request by mhutch**
This now makes it possible to test typing, triggering completion, and committing completion values

### CODE REVIEW

File: Microsoft.VisualStudio.MiniEditor/BaseViewImpl/MockFeatureServiceFactory.cs [hunk 1]
```diff
@@ -1,40 +0,0 @@
-using System;
-using System.ComponentModel.Composition;
-using Microsoft.VisualStudio.Utilities;
-
-namespace Microsoft.VisualStudio.MiniEditor.BaseViewImpl
-{
-	[Export (typeof (IFeatureServiceFactory))]
-	class MockFeatureServiceFactory : IFeatureServiceFactory
```
Comment (garuma) [inline Microsoft.VisualStudio.MiniEditor/BaseViewImpl/MockFeatureServiceFactory.cs:8]: Why remove this?

File: Microsoft.VisualStudio.MiniEditor/Microsoft.VisualStudio.MiniEditor.csproj [hunk 2]
```diff
@@ -33,286 +32,387 @@
...
+    <Compile Include="$(BaseEditorPath)Text\Util\TextUIUtil\VacuousTextViewModel.cs" />
+    <Compile Include="$(BaseEditorPath)Text\Util\TextUIUtil\VacuousTextDataModel.cs" />
   </ItemGroup>
   <ItemGroup>
     <EmbeddedResource Include="$(BaseEditorPath)Core\Impl\ContentType\Strings.resx">
       <Generator>ResXFileCodeGenerator</Generator>
-      <ManifestResourceName>Microsoft.VisualStudio.CoreUtilityImplementation.ContentType.Strings</ManifestResourceName>
-      <LastGenOutput>$(BaseEditorPath)Core\Impl\ContentType\Strings.Designer.cs</LastGenOutput>
+      <LogicalName>Microsoft.VisualStudio.CoreUtilityImplementation.ContentType.Strings.resources</LogicalName>
+      <LastGenOutput>vs-editor-api\src\Editor\Core\Impl\ContentType\Strings.Designer.cs</LastGenOutput>
```
Comment (garuma) [inline Microsoft.VisualStudio.MiniEditor/Microsoft.VisualStudio.MiniEditor.csproj:319]: Any reason to remove `$(BaseEditorPath)` from those paths?

File: Microsoft.VisualStudio.MiniEditor/BaseViewImpl/MockFeatureServiceFactory.cs (hunk 1 above)
Comment (garuma) [reply Microsoft.VisualStudio.MiniEditor/BaseViewImpl/MockFeatureServiceFactory.cs:8]: Nevermind https://github.com/garuma/MiniEditor/pull/4/commits/d1e4c8ae76563977f99f817991f7c1f5dc7da106

File: Microsoft.VisualStudio.MiniEditor/Microsoft.VisualStudio.MiniEditor.csproj (hunk 2 above)
Comment (mhutch) [reply Microsoft.VisualStudio.MiniEditor/Microsoft.VisualStudio.MiniEditor.csproj:319]: nope, was just VSMac doing it again :'(

**Pull request merged**